samples/peptide_mixture_grxmc_ideal.py

```diff
@@ -264,27 +263,41 @@
-HH_charge_dict = pmb.calculate_HH_Donnan(c_macro={peptide1: pep1_concentration, peptide2: pep2_concentration}, c_salt=c_salt, pH_list=pH_range_HH)
-Z_HH_Donnan = HH_charge_dict["charges_dict"]
-pH_sys = HH_charge_dict["pH_system_list"]
-xi = HH_charge_dict["partition_coefficients"]
-
-fig, ax = plt.subplots(figsize=(10, 7))
-plt.errorbar(pH_range, np.asarray(Z_pH)/N_peptide1_chains, yerr=np.asarray(err_Z_pH)/N_peptide1_chains, fmt = 'o', capsize=3, label='Simulation')
-ax.plot(pH_range_HH, np.asarray(Z_HH_Donnan[peptide1])+np.asarray(Z_HH_Donnan[peptide2]), "--r", label='HH+Donnan')
-plt.legend()
-plt.xlabel('pH')
-plt.ylabel('Charge of the peptide 1 + peptide 2 / e')
-plt.show()
-plt.close()
-
-fig, ax = plt.subplots(figsize=(10, 7))
-plt.errorbar(pH_range, xi_plus, yerr=err_xi_plus, fmt = 'o', capsize=3, label='Simulation')
-ax.plot(pH_range_HH, np.asarray(xi), "-k", label='HH+Donnan')
-plt.legend()
-plt.xlabel('pH')
-plt.ylabel(r'partition coefficient $\xi_+$')
-plt.show()
-plt.close()
+
+if args.mode == 'standard' or args.mode == 'unified':
+    # Calculate the ideal titration curve of the peptide with Henderson-Hasselbach equation
+    pH_range_HH = np.linspace(2, 12, num=100)
+    HH_charge_dict = pmb.calculate_HH_Donnan(c_macro={peptide1: pep1_concentration, peptide2: pep2_concentration}, c_salt=c_salt, pH_list=pH_range_HH)
+    Z_HH_Donnan = HH_charge_dict["charges_dict"]
+    xi = HH_charge_dict["partition_coefficients"]
+
+    # Plot the results
+    fig, ax = plt.subplots(figsize=(10, 7))
+    plt.errorbar(pH_range, np.asarray(Z_pH)/N_peptide1_chains, yerr=np.asarray(err_Z_pH)/N_peptide1_chains, fmt = 'o', capsize=3, label='Simulation')
+    ax.plot(pH_range_HH, np.asarray(Z_HH_Donnan[peptide1])+np.asarray(Z_HH_Donnan[peptide2]), "--r", label='HH+Donnan')
+    plt.legend()
+    plt.xlabel('pH')
+    plt.ylabel('Charge of the peptide 1 + peptide 2 / e')
+    plt.show()
+    plt.close()
+
+    fig, ax = plt.subplots(figsize=(10, 7))
+    plt.errorbar(pH_range, xi_plus, yerr=err_xi_plus, fmt = 'o', capsize=3, label='Simulation')
+    ax.plot(pH_range_HH, np.asarray(xi), "-k", label='HH+Donnan')
+    plt.legend()
+    plt.xlabel('pH')
+    plt.ylabel(r'partition coefficient $\xi_+$')
+    plt.show()
+    plt.close()
+
+elif args.mode == 'standard-test' or args.mode == 'unified-test':
+    # Calculate the ideal titration curve of the peptide with Henderson-Hasselbach equation
+    HH_charge_dict = pmb.calculate_HH_Donnan(c_macro={peptide1: pep1_concentration, peptide2: pep2_concentration}, c_salt=c_salt, pH_list=pH_range)
+    Z_HH_Donnan = HH_charge_dict["charges_dict"]
+    xi = HH_charge_dict["partition_coefficients"]
+    
+    # Check if charges agree
+    np.testing.assert_allclose(np.asarray(Z_pH)/N_peptide1_chains, np.asarray(Z_HH_Donnan[peptide1])+np.asarray(Z_HH_Donnan[peptide2]), rtol=0.01, atol=0.05)
+
+    # Check if partition coefficients agree
+    np.testing.assert_allclose(xi_plus, np.asarray(xi), rtol=0.1, atol=0.1)
```

Comment:
I would move this lines to `testsuite/grxmc_ideal_tests.py` . In general, I would avoid test-specific in the sample scripts as much as possible because they are meant to be for the users to learn and I think that it will only confuse them.